clients/drcachesim/tests/drcachesim_unit_tests.cpp

```diff
@@ -235,6 +235,64 @@ unit_test_compulsory_misses()
                                       cache_split_t::INSTRUCTION) == 6);
 }
 
+void
+unit_test_child_hits()
+{
+    // Ensure child hits include all lower levels.
+    std::string config = R"MYCONFIG(// 3-level simple config.
+num_cores       1
+line_size       64
+L1I {
+  type            instruction
+  core            0
+  size            256
+  assoc           4
+  prefetcher      none
+  parent          L2
+}
+L1D {
+  type            data
+  core            0
+  size            256
+  assoc           4
+  prefetcher      none
+  parent          L2
+}
+L2 {
+  size            8K
+  assoc           8
+  inclusive       true
+  prefetcher      none
+  parent          LLC
+}
+LLC {
+  size            1M
+  assoc           8
+  inclusive       true
+  prefetcher      none
+  parent          memory
+}
+)MYCONFIG";
+    std::istringstream config_in(config);
+    cache_simulator_t cache_sim(&config_in);
+
+    memref_t ref;
+    ref.data.type = TRACE_TYPE_READ;
+    ref.data.size = 1;
+
+    const int num_accesses = 16;
+    for (int i = 0; i < num_accesses; i++) {
+        ref.data.addr = 64 + i;
+        if (!cache_sim.process_memref(ref)) {
+            std::cerr << "drcachesim unit_test_child_hits failed: "
+                      << cache_sim.get_error_string() << "\n";
+            exit(1);
+        }
+    }
+    assert(cache_sim.get_cache_metric(metric_name_t::CHILD_HITS, 3, 0) ==
```

Comment:
For a slightly stronger check, maybe add  `&& cache_sim.get_cache_metric(metric_name_t::CHILD_HITS, 1, 0)`

Reply:
Added several more stats asserts.